src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -120,6 +123,38 @@ private void InitializeFileWatchers()
                         TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously)));
         }
 
+        private void StopFileWatchers()
+        {
+            lock (_stopWatchersLock)
```

Comment:
While at it, would you mind also adding a log entry here saying that file watchers are being stopped?

There were a couple of scenarios where FileWatchers were disposed and it was hard to figure out when debugging through logs.